crates/driver/src/logic/competition/solution/settlement.rs

```diff
@@ -0,0 +1,82 @@
+use {
+    super::Approvals,
+    crate::{
+        boundary,
+        logic::{competition, eth},
+        node,
+        EthNode,
+        Simulator,
+    },
+    num::{BigRational, ToPrimitive},
+};
+
+#[derive(Debug)]
+pub struct Settlement(boundary::Settlement);
+
+/// The solution score. This is often referred to as the "objective value".
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub struct Score(BigRational);
+
+impl From<Score> for f64 {
+    fn from(score: Score) -> Self {
+        score.0.to_f64().expect("value can be represented as f64")
+    }
+}
+
+impl Settlement {
+    /// Encode a solution into an onchain settlement.
+    pub async fn encode(
+        node: &EthNode,
+        _auction: &competition::Auction,
+        solution: competition::Solution,
+    ) -> Self {
+        let mut settlement = boundary::Settlement::new(solution.prices);
+        // TODO No unwrap
+        let approvals = Self::filter_approvals(node, &solution.approvals)
+            .await
+            .unwrap();
+        for approval in approvals {
+            settlement
+                .encoder
+                .append_to_execution_plan(boundary::Approval::Approve {
+                    token: approval.spender.token.0,
+                    spender: approval.spender.address.0,
+                });
+        }
+        // TODO Encode the remaining executions, I believe the auction is needed for
+        // this
```

Comment:
I think it may also need order data (which is included in the auction, but is not sent in complete form to the solvers - things like `appData` have no bearing on how an order executes, but is required for encoding settlement calldata because of the signature verification).

So I do think `auction` is needed here as well.